src/pdf_parser/pdf_utils/disambiguate_layout.py

```diff
@@ -234,6 +249,19 @@ def reduce_all_overlapping_boxes(
         for j, box_2 in enumerate(blocks):
             if i == j:
                 continue
+            # Check if the boxes fully overlap in either direction. This handles edge cases
```

Comment:
I would just pass the boxes into the function as the argument and put these comments in the doc string